domain-services/fiscal-event-post-processor/src/main/java/org/egov/service/FiscalEventDereferenceService.java

```diff
@@ -22,13 +27,124 @@
     @Autowired
     private FiscalEventDereferenceEnrichmentService enricher;
 
+    @Autowired
+    private CoaUtil coaUtil;
+
+    @Autowired
+    private GovernmentUtil governmentUtil;
+
+    @Autowired
+    private ProjectUtil projectUtil;
+
+    @Autowired
+    private EatUtil eatUtil;
+
+    @Autowired
+    private DepartmentUtil departmentUtil;
+
     public void dereference(FiscalEventRequest fiscalEventRequest) {
         FiscalEventDeReferenced fiscalEventDeReferenced =  new FiscalEventDeReferenced();
+        dereferenceTenantId(fiscalEventRequest,fiscalEventDeReferenced);
         dereferenceCoaId(fiscalEventRequest,fiscalEventDeReferenced);
-        enricher.enrich(fiscalEventDeReferenced,fiscalEventDeReferenced);
+        dereferenceProjectId(fiscalEventRequest, fiscalEventDeReferenced);
+        enricher.enrich(fiscalEventRequest,fiscalEventDeReferenced);
         producer.push(processorConfig.getEventProcessorMongoDB(),fiscalEventDeReferenced);
```

Comment:
This service will return FiscalEventDeReferenced. Write the producer.push statement in the consumer.

Reply:
As per my point of view  consumer should only consume data . service will take care of producing and other business logic on it. but anyway I'll change accordingly.